src/controller.c

```diff
@@ -84,10 +109,63 @@ void boss_mode() {
     clear();
 }
 
+void pause_mode(State* s) {
+    mvprintw(20, GRID_W+5, "*** PAUSED ***");
+
+    // wait until game resumed
+    while (getch() != RESUME_KEY);
```

Comment:
I think boss mode has this line too. Can you refactor this into a small method `wait_until_resumed`. It also takes care of the comments